tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -251,11 +251,14 @@ public void onFailure(Throwable e) {
             private void doRequestEndProcessing() {
                 if (span != null) {
                     span.finish();
+                    TracingContext tracingContext = topChannel.getTracingContext();
```

Comment:
it would be initialized via `TChannel.Builder.setExecutorService()`. the latter would be called by `XTChannel.setupTChannel()` [TODO] or by whoever sets up the `TChannel` instance with thread-local-bound trace implementation.